c7n/filters/iamaccess.py

```diff
@@ -161,22 +164,22 @@ def handle_conditions(self, s):
 
         results = []
         for c in conditions:
-            results.append(self.handle_condition(s, c))
+            results.append(self.handle_condition(conditions, c))
 
         return all(results)
 
-    def handle_condition(self, s, c):
+    def handle_condition(self, conditions, c):
```

Comment:
changing this signature seems suspect, the intent was that in some contexts evaluating the condition requires the whole statement. effectively `conditions` is already present in the statement `s`. I'd rather not change the signature arbitrarily for a single resource permission case to have less information for other use cases.

Reply:
OK, do you agree though that the sns:endpoint handler should yield to the sns:sourceowner handler if present and whitelist_endpoints wasn't specified?

Reply:
passing the statement still has all the conditions.

Reply:
and yes re sourceowner, though ideally thats only if sourceowner id is validated against account whitelist ids.